### PR TITLE
Unbind hive config from iceberg and delta lake modules

### PR DESCRIPTION
**Pull request by homar**
## Description

> Is this change a fix, improvement, new feature, refactoring, or other?

> Is this a change to the core query engine, a connector, client library, or the SPI interfaces? (be specific)

> How would you describe this change to a non-technical end user or system administrator?

## Documentation

( ) No documentation is needed.
( ) Sufficient documentation is included in this PR.
( ) Documentation PR is available with #prnumber.
( ) Documentation issue #issuenumber is filed, and can be handled later.

## Release notes

( ) No release notes entries required.
( ) Release notes entries required with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

**Comment by findepi** [comment]
let's review this once https://github.com/trinodb/trino/pull/12595 is merged

**Comment by findepi** [comment]
please rebase

**Comment by homar** [comment]
> please rebase

yes i know i was waiting for the [other one](https://github.com/trinodb/trino/pull/12595)

File: plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergPlugin.java [hunk 1]
```diff
@@ -31,6 +31,8 @@
 
 public class TestIcebergPlugin
 {
+    private static final String HIVE_VIEWS_ENABLED = "hive.hive-views.enabled";
```
Comment (findepi) [inline plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergPlugin.java:34]: left over

**Comment by colebow** [comment]
We're going to need to include docs for this, too. @homar can you get a PR up today, or do you want me to? Release is today, so it'll need to be a quick turnaround.

cc @findepi

**Comment by homar** [comment]
@colebow PR for docs change https://github.com/trinodb/trino/pull/12751 
Sorry it took me so long. I had busy afternoon

**Pull request merged**